### ISSUE TITLE
why val in train.py use 'ema.ema' instead of 'model'?

### ISSUE THREAD

**Issue by Skyninth** [ultralytics/yolov5#9930]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I find process val in train.py use the model of 'ema.ema' instead of 'model', like this

                    results, maps, _ = validate.run(data_dict,
                                                    batch_size=batch_size // WORLD_SIZE * 2,
                                                    imgsz=imgsz,
                                                    half=amp,
                                                    model=ema.ema,
                                                    single_cls=single_cls,
                                                    dataloader=val_loader,
                                                    save_dir=save_dir,
                                                    plots=False,
                                                    callbacks=callbacks,
                                                    compute_loss=compute_loss)

I test 'model = deepcopy(model).eval()' , then I find the MAP is lower than ema.ema a lot, such as:
             model-Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95
                 all       1539       1656      0.764      0.517      0.553      0.262     
             ema-Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95
                 all       1539       1656      0.754      0.582      0.613      0.319    
but finally save the model is 'deepcopy(model).half()' . I want to know if there is some problem? Why I use val.py test the final saved model' MAP is same as ema.eam?But in train.py is not same?

In addition, I try use yolov5 to semi-supervised od. The initial teacher model and student model should be consistent, so should 'model' be selected as the initial model, or 'ema.ema' as the initial model?
